RELEASES.md

```diff
@@ -1,64 +1,89 @@
 # APM Server release checklist
 
-## Before the release
+The APM Server follows the Elastic Stack release schedule and versions.
+A release starts with a Feature Freeze period, during which only bug fixes
+are allowed to be merged into the specific release branch.
+We generally follow [semver](https://semver.org/) for release versions.
+For major and minor releases a new branch is cut off the main branch.
+For patch releases, only the version on the existing major and minor version branch is updated.
 
-* Backport all relevant commits
+## Feature Freeze
 
-  e.g. start with following command to see which files differ that are not necessarily expected:
-  `git diff elastic/main --name-only | grep -v '^vendor/' | grep -v '^docs/' | grep -v '^_beats'`
+* For patch releases, ensure all relevant backport PRs are merged. 
+  We use backport labels on PRs and automation to ensure labels are set.
 
-* Update beats
+## Day after Feature Freeze
+* For minor releases, cut a new release branch from `main` and update them.
 
-  run `make update-beats` in the branch from which the new branch will be created before FF to recognize potential issues
+  * Release branch: 
+    Update versions and ensure that the `BEATS_VERSION` in the Makefile is updated,
+    e.g. [#2803](https://github.com/elastic/apm-server/pull/2803/files).
+    Trigger a new beats update, once the beats branch is also created.
 
-* Update Changelog
```

Comment:
I think this step is still required. IMO, a better way to update changelogs file will be before we cut the FF branch. For example, for 8.4, before cutting the branch we can move the `head.asciidoc` -> `8.4.asciidoc` and  then after cutting the 8.4 branch, we can add a new `head.asciidoc` on main. WDYT?

Reply:
Moved the _Update Changelog_ section up, but phrased it to just move the content of `head` to the new `release version` docs file and then remove the `head` file from the release branch in the follow up.

Some docs build might depend on the `heads` file in main, so I didn't want to remove it from there.